usr/lib/python3/dist-packages/stprint/stprint.py

```diff
@@ -0,0 +1,336 @@
+#!/usr/bin/env python3
+
+## SPDX-FileCopyrightText: 2025 Benjamin Grande M. S. <ben.grande.b@gmail.com>
+## SPDX-FileCopyrightText: 2025 ENCRYPTED SUPPORT LLC <adrelanos@whonix.org>
+##
+## SPDX-License-Identifier: AGPL-3.0-or-later
+
+"""
+Sanitize text to be safely printed to the terminal.
+"""
+
+import textwrap
+import argparse
+import re
+import sys
+from typing import (
+    Generator,
+    Optional,
+    Pattern,
+    Union,
+)
+
+
+def exclude_pattern(original_pattern: str,
+                    negate_pattern: list[str]) -> str:
+    """Exclude matching next expression if provided expression matches.
+
+    Parameters
+    ----------
+    original_pattern : str
+        Pattern which contains unwanted matches.
+    negate_pattern : list[str]
+        Pattern to exclude from the next match.
+
+    Returns
+    -------
+    str
+        Regular expression with negative lookahead.
+
+    Examples
+    --------
+    Redact unsafe sequences by default:
+    >>> color_pattern = exclude_pattern(r"(30|31)", ["31"])
+    '(?!(?:31))(30|31)'
+    """
+    exclude_pattern_str = "|".join(map(re.escape, negate_pattern))
+    exclude_pattern_str = rf"(?!(?:{exclude_pattern_str}))"
+    return rf"{exclude_pattern_str}{original_pattern}"
+
+
+def get_color_pattern(colors: Optional[bool],
+                      extra_colors: Optional[bool],
+                      exclude_colors: Optional[list[str]]
+                      ) -> Pattern[str]:
+    """Print compiled RegEx for SGR sequences
+
+    The SGR implementation is well documented[1][2][3], here is an extract from
+    the Linux manual console_codes(4):
+
+    > The ECMA-48 SGR sequence ESC [ parameters m sets display attributes.
+    > Several attributes can be set in the same sequence, separated by
+    > semicolons. An empty parameter (between semicolons or string initiator or
+    > terminator) is interpreted as a zero.
+
+    But the above is vague, the syntax validation is not strict and assumptions
+    were made based on tests using XTerm on Linux, complementing the
+    definition:
+
+    *   Invalid SGR will lead to one of the following behaviors:
+        *   Invalidate the whole sequence skipping all SGR.
+        *   Invalidate only itself, skipping evaluation of that specific SGR.
+    *   Specifying a color multiple times will print only the last color
+        definition.
+    *   Specifying an attribute multiple times will combine them unless they
+        revert each other.
+
+    With the definitions above, we set the following pseudo regex:
+
+    -  4-bit SGR: (;*#(;#)*)?m
```

Comment:
It would be helpful if it was explicitly pointed out that `#` is a placeholder for an `sgr_4bit`. This bit of documentation is extremely useful now that I've managed to wrap my mind around the escape codes here, but the pound symbols confused me a lot to begin with.